### PR TITLE
Use email address when FROM field doesn't have a name.

### PR DESCRIPTION
**Pull request by protich**
Addresses issue #519.

### CODE REVIEW

File: include/class.mailfetch.php [hunk 1]
```diff
@@ -372,16 +372,20 @@ function createTicket($mid) {
...
         $vars['message']=Format::stripEmptyLines($this->getBody($mid));
         $vars['header']=$this->getHeader($mid);
         $vars['emailId']=$emailId?$emailId:$ost->getConfig()->getDefaultEmailId(); //ok to default?
-        $vars['name']=$vars['name']?$vars['name']:$vars['email']; //No name? use email
         $vars['mid']=$mailinfo['mid'];
 
-        if(!$vars['message']) //An email with just attachments can have empty body.
+        //Missing FROM name  - use email address.
+        if(!$vars['name'])
+            $vars['name'] = $vars['email'];
```
Comment (greezybacon) [inline include/class.mailfetch.php:385]: How is this different from original line 381?

**Comment by greezybacon** [comment]
Personally, I feel like this should be changed in the validation in the `api.tickets.php` rather than monkey patching the mail parsing equipment. Technically, if an email doesn't have a named from address, then it doesn't have a named from address.

In other words, to me, it seems to raise the question if the 'name' field is really required. It seems we're working around incorrect requirements rather than making sense of the input.

File: include/class.mailfetch.php (hunk 1 above)
Comment (protich) [inline include/class.mailfetch.php:385]: Just moved it down to group what we're monkey patching together - e.g empty message.

**Comment by protich** [comment]
@greezybacon - I agree with you to some extend - but the name patch is specific to email, not all other formats (xml/json) or even web need it. We have to be **very** forgiving when it comes to emails - e.g empty message.

**Comment by greezybacon** [comment]
I hear what you're saying, but why are creating two or more faces to the ticketing system? If an empty message is allowed via email but not via the web interface (staff or client), what are really requiring? We're truly just penalizing folks who submit tickets via the web. And, as a user, if I get fed up with the web interface, I'll just email in crappy emails with no name or message, and the ticketing system is suddenly just fine with missing data.

It's the same system. Why does the mode of input suddenly dictate what data is quote "required" by the system. I personally would vote for a return email stating validation errors to the email sender or make the same data interpretations on the web interface, if the name can technically dup the email address, and a ticket can be created without a message.

**Comment by protich** [comment]
No name email or empty message is not crappy email - it's allowed as per email specs. Should email specs dictate how we develop the rest of the software? I would say **no**. 

Email is just not a good medium to return errors - given how SMTP works. If we're going to accept emails then we have to be liberal - no way around it - good example is OutLook formatting.

**Comment by greezybacon** [comment]
Please understand, I see the issue, and I (would like to think I) understand your argument.

What I'm saying is this: (with a submission to the ticketing system with no name and no message): 
- via the staff web interface: **fail**: required fields not given
- via the client web interface: **fail**: required fields not given
- via the HTTP API: **fail**: required fields not given
- via email: no problem. Message becomes `(EMPTY)` and name dups the email field

What I'm suggesting is that the email interface mimic the rest of the interfaces. A return email could/should be sent indicating the ticketing system's refusal to create a ticket based on the email. The same process and feedback that would be represented via the other interfaces. This would be like a bounce email from the email system when the destination mailbox cannot be reached.

What I'm trying to convey is that the interfaces be consistent. If a message is not required, then we should just support creating a ticket with just a subject and an empty thread. If the name field is not truly required, then we should make it optional.

-- That's all I'm saying --

_PS_ - Dynamic forms worsen this issue, because they add even more fields that cannot necessarily be represented in MIME headers.

**Comment by protich** [comment]
I understand what you are saying and the overall goal. I'll be totally with you, if osTicket was M2M application. 

We'll have to just agree that not ALL interfaces are made equal in this case. Should the emailing end user even care or know that an organization is using osTicket in the backend? 

The difference, in my opinion, is - admin/installer can/should controls web forms and API (json/xml) interfaces - but not email delivery. Bouncing "deliverable" emails back to the **end users** - just because it's missing a name or message is just way too much - 1) osTicket doesn't control MTA to cleanly bounce incase of IMAP/POP 2) Emailing end user doesn't have ability to correct and send the email - the mail server has already accepted the email. 3) The said errors have nothing to do with email delivery.

Dynamic forms/fields illustrates my view point - you might as well not add extra fields if email interface will dictate what we allow. 

PS: Think of end users as typical mom and pap shop customers.
PSS: It also helps to think of email integration as a non-core optional supplemental interface

**Pull request merged**